frontend/src/components/dialogs/AssetCompareDialog.vue

```diff
@@ -0,0 +1,156 @@
+         * @param {{flows: { flows :[]}}} v1Snapshot - A snapshot object as the base for comparison
+         * @param {[{label: String, value: String}]} snapshotList - A list of snapshots to compare against where label is the snapshot name and value is the snapshot id
+         */
+            show (v1Snapshot, snapshotList) {
+                this.mode = 'compare'
+                this.payload = v1Snapshot
+                this.compareSnapshot = null
+                this.changes = []
+                this.changeIndex = 0
+                this.compareSnapshotList = snapshotList
+                this.$refs.dialog.show()
+            }
+        }
+    },
+    data () {
+        return {
+            payload: [],
+            snapshotList: [],
+            compareSnapshot: null,
+            compareSnapshotList: [],
+            mode: 'view', // view, compare
+            changes: [],
+            changeIndex: 0
+        }
+    },
+    computed: {
+        flow () {
+            return this.payload?.flows?.flows || []
+        },
+        header () {
+            return this.payload?.name || this.title || 'Flow'
+        }
+    },
+    mounted () {
+    },
+    methods: {
+        confirm () {
+            this.cleanup()
+            this.$refs.dialog.close()
+        },
+        renderFlows () {
+            this.cleanup()
+            const flowRenderer = new FlowRenderer()
+            flowRenderer.renderFlows(this.flow, {
+                container: this.$refs.compareViewer
+            })
+        },
+        async renderComparison (snapshotId) {
+            this.cleanup()
+            const compareSnapshot = await SnapshotsApi.getFullSnapshot(this.compareSnapshot)
+            if (!compareSnapshot?.flows?.flows) {
+                Alerts.emit('Flows not found in the selected snapshot', 'warning')
+                return
+            }
+            const flowRenderer = new FlowRenderer()
+            const flows = [this.flow, compareSnapshot?.flows?.flows]
+            const result = flowRenderer.compare(flows, {
+                container: this.$refs.compareViewer
+            })
+            this.changes = result?.changes || []
+        },
+        gotoNextDifference () {
+            this.changeIndex = (this.changeIndex + 1) % this.changes.length
+            this.changes[this.changeIndex].highlight()
+        },
+        gotoPreviousDifference () {
+            this.changeIndex = (this.changeIndex - 1 + this.changes.length) % this.changes.length
+            this.changes[this.changeIndex].highlight()
+        },
+        cleanup () {
+            while (this.$refs.compareViewer?.firstChild) {
+                this.$refs.compareViewer.removeChild(this.$refs.compareViewer.firstChild)
+            }
+            while (this.$refs.compareViewer?.firstChild) {
+                this.$refs.compareViewer.removeChild(this.$refs.compareViewer.firstChild)
+            }
```

Comment:
```suggestion
            while (this.$refs.compareViewer?.firstChild) {
                this.$refs.compareViewer.removeChild(this.$refs.compareViewer.firstChild)
            }
```

Am I missing something? Why twice?